packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -234,7 +248,7 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
 
     const newHistoricalPricePeriods = await this._getOhlcPricePeriods(earliestHistoricalTimestamp, currentTime);
     const newPrice = this.twapLength
-      ? this._computeTwap(currentTime, newHistoricalPricePeriods)
+      ? this._computeTwap(currentTime, newHistoricalPricePeriods, this.twapLength)
```

Comment:
Do we not consider ancillary data in the update function? Is that a problem? Shouldn't the current price and the historical price be computed the same way?

Or is the assumption that there are no contracts that use current price that also have ancillary data? i.e. all ancillary data requests go through the OO.

Reply:
> Or is the assumption that there are no contracts that use current price that also have ancillary data? i.e. all ancillary data requests go through the OO.

This is the assumption. Also, currently the `update()` function is not designed to accept any ancillary data. How would that work you think? I do think we should refactor `getCurrentPrice` to take in `ancillaryData` soon

Reply:
This issue raised by @Reinis-FRP is related: https://github.com/UMAprotocol/protocol/pull/3659#issuecomment-984988812

How do we connect any `twapLength` specified in `ancillaryData` to the `update` method in the price feed?

Reply:
We could add an optional input param to `update` for `ancillaryData` so that anyone can "update" it for a specific `ancillaryData`. Is that reasonable..?

Reply:
What about providing an extra arg to getPrice()?

Reply:
The reason I say this is that it seems a little difficult to reason about if you pass in the ancillary data into update: whenever you call getPrice, you need to know what ancillary data was passed in the last time someone called update. Depending on where we call update, the ancillary data could have been left blank or could have been different. This means that you _always_ just need to call update() and immediately follow with getPrice().

This isn't a strong opinion, but I think a more natural way to set values like this is either:
1. In the constructor and have it be immutable.
2. As an argument to getPrice, similar to getHistoricalPrice.

I'm guessing it's difficult to remove it from update() because there are async calls in there that rely on it. Maybe the right solution is to leave this as you have it for now, but add a task to drop update entirely and make getPrice async?

Reply:
> whenever you call getPrice, you need to know what ancillary data was passed in the last time someone called update. Depending on where we call update, the ancillary data could have been left blank or could have been different. This means that you always just need to call update() and immediately follow with getPrice().

This is an assumption we make for how `getCurrentPrice` works though, so I think this is very reasonable. `getCurrentPrice` will always return an unexpected price if the caller does not know when the last time `update` was called. 

However, I could see us adding `ancillaryData` to the constructor params. 

> the right solution is to leave this as you have it for now, but add a task to drop update entirely and make getPrice async for all price feeds

This 100% is true but we should save for future big PR

Reply:
ye, I agree this is not ideal and a but clunky in some cases. it's very much strapped onto our existing implementation. 

I agree with both of you what an ideal future version of this would look like, but it should totally be ignored for the time being given how much work those changes would require.